### PR TITLE
VB-3779, VB-4052, VB-4053 Handle unknown booker signing in

### PR DESCRIPTION
**Pull request by tpmcgowan**
* Add a new 'Access denied' page for when a user is authenticated successfully via GOV.UK One Login but the do not have a booker account
  * Update test helpers to handle mocking this scenario
* Update the content and path of the default 'auth error' page

### CODE REVIEW

File: server/middleware/populateCurrentBooker.ts [hunk 1]
```diff
@@ -24,7 +25,7 @@ export default function populateCurrentBooker(bookerService: BookerService): Req
     } catch (error) {
       if (error.status === 404) {
         logger.info(`Failed to retrieve booker reference for: ${res.locals.user.sub}`)
-        return res.redirect('/autherror')
+        return req.path === paths.ACCESS_DENIED ? next() : res.redirect('/access-denied')
```
Comment (tpmcgowan) [inline server/middleware/populateCurrentBooker.ts:28]: Need this check otherwise there's a redirect loop

**Review by hutcheonb-moj — APPROVED** [review]
LGTM

File: server/routes/accessDeniedController.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+export default class AccessDeniedController {
+  public constructor() {}
+
+  public view(): RequestHandler {
+    return async (req, res) => {
+      if (req.session.booker) {
+        return res.redirect(paths.HOME)
+      }
+
+      return res.render('accessDenied')
```
Comment (psoleckimoj) [inline server/routes/accessDeniedController.ts:13]: I feel I'm getting slightly obsessed by ternaries but nothing wrong with this at all 😄 

`return req.session.booker ? res.redirect(paths.HOME) : res.render('accessDenied')`

**Review by psoleckimoj — DISMISSED** [review]
LGTM!

File: server/routes/accessDeniedController.ts (hunk 2 above)
Comment (tpmcgowan) [reply server/routes/accessDeniedController.ts]: Good suggestion; looks neater now. Have updated!

**Review by psoleckimoj — APPROVED** [review]
LGTM!

**Pull request merged**